packages/camera/camera_platform_interface/lib/src/types/video_stabilization_mode.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// The possible video stabilization modes that can be capturing video.
+enum VideoStabilizationMode {
+  /// Video stabilization is disabled.
+  off,
+
+  /// Basic video stabilization is enabled.
+  ///
+  /// Maps to CONTROL_VIDEO_STABILIZATION_MODE_ON on Android
+  /// and throws CameraException on iOS.
```

Comment:
Related to my comment about `standard`, it seems very surprising that `on` would throw on iOS since clearly stabilization can be on.

Can we make a more generic set of options that can be more easily mapped to various implementations? Having semi-disjoint options across two platforms is a sign that this is not generic enough, and would thus get even more confusing if a third implementation were added later.

Reply:
I wonder if a first approach to solving the mapping issue doesn't start by doing something as simple as cleaning up the comments. For instance:

```dart
  /// Extended cinematic video stabilization is enabled.
  ///
  /// Maps to AVCaptureVideoStabilizationModeCinematicExtended on iOS and
  /// throws CameraException on Android.
  cinematicExtended;
```
Saying that it throws on Android is kind of redundant, because until the day it gets implemented on Android, it will never be included in the list returned by ```getVideoStabilizationSupportedModes()```. It's just that the comment is hinting the user on what the current implementation is doing. So, an easy thing to do would be to change it to:

```dart
  /// Extended cinematic video stabilization is enabled.
  ///
  /// Maps to AVCaptureVideoStabilizationModeCinematicExtended on iOS.
  cinematicExtended;
```

Next, does it make sense to keep the "Maps to AVCaptureVideoStabilizationModeCinematicExtended on iOS"? On one hand, it doesn't because it will be incomplete if other platforms get an implementation. It might mislead the user into thinking that only iOS still has an implementation even when it ceases to be exclusive to iOS. On the other hand, it is very convenient for the user to know how it maps to iOS just by looking at the enum (which is something that I do a lot with ResolutionPreset and to a lesser degree with ImageFormatGroup. So, I don't really know what the best approach would be.

Reply:
> The 'Standard' mode maps to what I considered to be the standard video stabilization on each platform, so I mapped it to CONTROL_VIDEO_STABILIZATION_MODE_PREVIEW_STABILIZATION and AVCaptureVideoStabilizationModeStandard.

Can you elaborate on why preview stabilization mode is what you would consider "standard" on Android, when from the enum naming it seems like the developers of the Android API would consider "on" to be the standard mode? Is it about matching iOS behaviors?

In practice, what is the behavioral difference between `CONTROL_VIDEO_STABILIZATION_MODE_ON ` and `AVCaptureVideoStabilizationModeStandard`, and between `CONTROL_VIDEO_STABILIZATION_MODE_PREVIEW_STABILIZATION` and `AVCaptureVideoStabilizationModeStandard`?

Part of the issue here I think is that the Android constant names communicate (albeit not very clearly) a behavioral difference, whereas the names you've chosen to map those two sound indistinguishable to me, so as a client I would have no idea what the difference between `on` and `standard` is. If the difference between them is *only* whether or not the preview is stabilized, then the names should reflect that, rather than hide it. (If that distinction doesn't exist on iOS it's okay to have a constant that doesn't exist on iOS, but it doesn't make sense for something called `on` not to exist on iOS.)

> So, yes, I am open to making changes, but it would be helpful if you could suggest the concrete mapping, including descriptions

I don't know what a better mapping is without more information; the spreadsheet contains the docs information, but that doesn't answer important questions about how they actually compare. E.g., I don't know from the spreadsheet what the preview behavior is on iOS.

What I do know is that I find the current names *actively confusing* as a naive client of the API, which is a red flag.

Reply:
> Next, does it make sense to keep the "Maps to AVCaptureVideoStabilizationModeCinematicExtended on iOS"? On one hand, it doesn't because it will be incomplete if other platforms get an implementation. It might mislead the user into thinking that only iOS still has an implementation even when it ceases to be exclusive to iOS. On the other hand, it is very convenient for the user to know how it maps to iOS just by looking at the enum (which is something that I do a lot with ResolutionPreset and to a lesser degree with ImageFormatGroup. So, I don't really know what the best approach would be.

The approach we've taken in newer code is to frame the mapping as examples. It's still not ideal in that in theory the platform implementations could diverge over time because it's a different package, but we balance that against the value of people being able to see concretely what things will mean in practice. So it's fine to say something like:

> /// For example, this corresponds to:
> /// - AVFoo on iOS
> /// - CONTROL_VIDEO_BaR on Android